### PR TITLE
Granite 3.1

### PR DESCRIPTION
**Pull request by gabe-l-hart**
### Model Interaction

- [ ] **Flexible LLM platform support** The platform should be easily switchable. Use LangChain or LlamaIndex.
- [ ] **Use prompt guide corresponding to the model**  For example for [Granite 3.x Language Models](https://www.ibm.com/granite/docs/models/granite/#prompt-anatomy)

### Data

- [ ] **Example data**: Follow the example data guidance.

### Notebook requirements

- [x] **Notebook outputs cleared**: Ensure all notebook outputs are cleared.
- [ ] **Automated testing**: Add the recipe to the automated tests as described [here](https://github.com/ibm-granite-community/community/wiki/Automated-Validation)
- [ ] **Test in Google Colab**:
    - [ ] Test that it works in Google Colab (Python 3.10.12).
    - [ ] Colab has its own package set and Python version, so ensure compatibility.
- [ ] **Test locally**:
    - [ ] Ensure the code works in a fresh Python virtual environment (venv).
- [ ] **Standard access to secrets and variables** Include `!pip install git+https://github.com/ibm-granite-community/utils` in the first code cell in order to make `get_env_var` available to accessing secrets and variables in the recipe.

### Incoming References

- [ ] **README.md updates**:
    - [ ] Add a link to the recipe in the Table of Contents (ToC).
    - [ ] Include a Colab button after that link.

### GitHub

- [x] **Commits signed**: All commits must be GPG or SSH signed.
- [x] **DCO Compliance**: Developer Certificate of Origin (DCO) applies to the code, documentation, and any example data provided. Ensure commits are signed off.

### CODE REVIEW

File: recipes/Components/Langchain_Embeddings_Models.ipynb [hunk 1]
```diff
@@ -37,9 +37,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Commented out to avoid downloading the model.\n",
-    "# from langchain_huggingface import HuggingFaceEmbeddings\n",
-    "# embeddings_model = HuggingFaceEmbeddings(model_name=\"all-MiniLM-L6-v2\")"
+    "from langchain_huggingface import HuggingFaceEmbeddings\n",
+    "embeddings_model = HuggingFaceEmbeddings(model_name=\"ibm-granite/granite-embedding-278m-multilingual\")"
```
Comment (bjhargrave) [inline recipes/Components/Langchain_Embeddings_Models.ipynb]: Should this be 30m-english as it is similar in size/language to the embedding it is replacing?

File: recipes/Components/Langchain_LLMs.ipynb [hunk 2]
```diff
@@ -102,7 +102,7 @@
     "from ibm_granite_community.notebook_utils import get_env_var\n",
     "\n",
     "model = WatsonxLLM(\n",
-    "    model_id=\"ibm/granite-3-8b-instruct\",\n",
+    "    model_id=\"ibm/granite-3.1-8b-instruct\",\n",
```
Comment (bjhargrave) [inline recipes/Components/Langchain_LLMs.ipynb]:
```suggestion
    "    model_id=\"ibm/granite-3-8b-instruct\",\n",
```

They did not change the id.

File: recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb [hunk 3]
```diff
@@ -119,9 +119,13 @@
     "\n",
     "The Granite models available on WatsonX are listed on the [Resource Hub](https://dataplatform.cloud.ibm.com/samples?context=wx&tab=foundation-model&query=granite). For more information about Granite models on WatsonX, see [Foundation models on watsonx.ai](https://www.ibm.com/products/watsonx-ai/foundation-models).\n",
     "\n",
-    "[`ibm/granite-8b-code-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-8b-code-instruct?context=wx)\n",
+    "[`ibm/granite-3.1-8b-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3.1-8b-instruct?context=wx)\n",
```
Comment (bjhargrave) [inline recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb]:
```suggestion
    "[`ibm/granite-3-8b-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3-8b-instruct?context=wx)\n",
```

File: recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb [hunk 4]
```diff
@@ -119,9 +119,13 @@
...
     "\n",
-    "[`ibm/granite-3-8b-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3-8b-instruct?context=wx)"
+    "[`ibm/granite-3.1-2b-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3.1-2b-instruct?context=wx)\n",
```
Comment (bjhargrave) [inline recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb]:
```suggestion
    "[`ibm/granite-3-2b-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3-2b-instruct?context=wx)\n",
```

File: recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb [hunk 5]
```diff
@@ -119,9 +119,13 @@
...
+    "\n",
+    "[`ibm/granite-3.1-3b-a800m-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3.1-3b-a800m-instruct?context=wx)\n",
+    "\n",
+    "[`ibm/granite-3.1-1b-a400m-instruct`](https://dataplatform.cloud.ibm.com/wx/samples/models/ibm/granite-3.1-1b-a400m-instruct?context=wx)"
```
Comment (bjhargrave) [inline recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb]: These won't be on Watsonx until sometime in the future.

File: recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb [hunk 6]
```diff
@@ -131,11 +135,10 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# model_id = \"ibm/granite-3b-code-instruct\"\n",
-    "# model_id = \"ibm/granite-8b-code-instruct\"\n",
-    "# model_id = \"ibm/granite-20b-code-instruct\"\n",
-    "# model_id = \"ibm/granite-34b-code-instruct\"\n",
-    "model_id = \"ibm/granite-3-8b-instruct\""
+    "model_id = \"ibm/granite-3.1-8b-instruct\"\n",
+    "# model_id = \"ibm/granite-3.1-2b-instruct\"\n",
```
Comment (bjhargrave) [inline recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb]:
```suggestion
    "model_id = \"ibm/granite-3-8b-instruct\"\n",
    "# model_id = \"ibm/granite-3-2b-instruct\"\n",
```

File: recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb [hunk 7]
```diff
@@ -131,11 +135,10 @@
...
+    "# model_id = \"ibm/granite-3.1-3b-a800m-instruct\"\n",
+    "# model_id = \"ibm/granite-3.1-1b-a400m-instruct\""
```
Comment (bjhargrave) [inline recipes/Getting_Started/Getting_Started_with_WatsonX.ipynb]: These won't be on watsonx till later.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

**Comment by gabe-l-hart** [comment]
Thanks @bjhargrave! All comments resolved

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**